includes/class-wc-cart.php

```diff
@@ -1738,9 +1738,23 @@ public function add_discount( $coupon_code ) {
 
 		// If its individual use then remove other coupons
 		if ( $the_coupon->get_individual_use() ) {
-			$this->applied_coupons = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+			$coupons_to_keep = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+
+			foreach ( $this->applied_coupons as $applied_coupon ) {
+				$keep_key = array_search( $applied_coupon, $coupons_to_keep );
```

Comment:
Oh gotcha. That is Step 5. Those are going to be new coupons that didn't already exist in `$this->applied_coupons`. Having that keeps the backwards compatibility, since the previous versions could add new coupons using the `woocommerce_apply_individual_use_coupon` filter.